packages/contentful--app-scripts/lib/clean-up/clean-up-bundles.js

```diff
@@ -29,20 +32,67 @@ const scheduleBundleDeletion = async (bundlesToDelete, client, settings) => {
 };
 
 async function cleanUpBundles(settings) {
-  let bundles, definition;
+  let bundlesToDelete, definition;
+
   const client = createClient(
     {
       host: settings.host,
       accessToken: settings.accessToken,
     },
     { type: 'plain' }
   );
-  const bundlesSpinner = ora(`Fetching all bundles...`).start();
-  try {
-    bundles = await client.appBundle.getMany({
+
+  const fetchAppBundles = async (limit = DEFAULT_BUNDLES_TO_FETCH, skip = 0) => {
+    return await client.appBundle.getMany({
       appDefinitionId: settings.definition.value,
       organizationId: settings.organization.value,
+      query: {
+        limit,
+        skip,
+      },
     });
+  };
+
+  const getAppBundles = async (requestedAmount = DEFAULT_BUNDLES_TO_FETCH) => {
+    if (requestedAmount < 1) {
+      throw new Error('Requested amount of bundles to fetch must be greater than 0');
+    }
+    const getBundles = async (limit, skip) => {
+      const result = await fetchAppBundles(limit);
+      const currLength = skip + result.items.length;
+
+      if (currLength <= requestedAmount && result.total > currLength) {
+        return [...result.items, ...(await getBundles(result.items.length, currLength))];
+      } else {
+        return result.items;
+      }
+    };
+
+    const all = await getBundles(DEFAULT_BUNDLES_TO_FETCH, 0);
+
+    return all.slice(0, requestedAmount);
```

Comment:
Shouldn't we only slice after sorting the list?

Reply:
hmm... Yeah. That's why I wanted the sorting from the API in the first place :D 
fix coming up